File: integrationtests/workspaces/go/another_consumer.go
package main

import "fmt"

// AnotherConsumer is a second consumer of shared types and functions
func AnotherConsumer() {
	// Use helper function
	fmt.Println("Another message:", HelperFunction())

	// Create another SharedStruct instance
	s := &SharedStruct{
		ID:        2,
		Name:      "another test",
		Value:     99.9,
		Constants: []string{SharedConstant, "extra"},
	}

	// Use the struct methods
	if name := s.GetName(); name != "" {
		fmt.Println("Got name:", name)
	}

	// Implement the interface with a custom type
	type CustomImplementor struct {
		SharedStruct
	}

	custom := &CustomImplementor{
		SharedStruct: *s,
	}

	// Custom type implements SharedInterface through embedding
	var iface SharedInterface = custom
	iface.Process()

	// Use shared type as a slice type
	values := []SharedType{1, 2, 3}
	for _, v := range values {
		fmt.Println("Value:", v)
	}
}

File: integrationtests/workspaces/rust/src/another_consumer.rs
// Another consumer module for testing references
use crate::helper::helper_function;
use crate::types::{
    SharedInterface, SharedStruct, SharedType, SHARED_CONSTANT,
};

pub fn another_consumer_function() {
    // Use the helper function
    let result = helper_function();
    println!("Helper result from another consumer: {}", result);

    // Use shared struct
    let s = SharedStruct::new("another test");
    println!("Struct in another consumer: {}", s.name);

    // Use shared interface
    let _iface: &dyn SharedInterface = &s;
    
    // Use shared constant
    println!("Constant in another consumer: {}", SHARED_CONSTANT);

    // Use shared type
    let _t: SharedType = String::from("another test");
}